Add support for arm

while compiling for arm, got the following errors:

<details> <summary> build errors </summary>

```
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:117:38
    |
117 |     ffi::duk_get_prop_string(ctx, 0, ERROR_KEY.as_ptr());
    |                                      ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:121:38
    |
121 |     ffi::duk_put_prop_string(ctx, 0, ERROR_KEY.as_ptr());
    |                                      ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:157:43
    |
157 |         ffi::duk_put_prop_string(ctx, -2, ERROR_KEY.as_ptr());
    |                                           ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:167:43
    |
167 |         ffi::duk_get_prop_string(ctx, -1, ERROR_KEY.as_ptr());
    |                                           ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:171:43
    |
171 |         ffi::duk_put_prop_string(ctx, -2, ERROR_KEY.as_ptr());
    |                                           ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:260:37
    |
260 |     ffi::duk_put_global_string(ctx, UDATA.as_ptr());
    |                                     ^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:264:37
    |
264 |     ffi::duk_put_global_string(ctx, ANYMAP.as_ptr());
    |                                     ^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:276:37
    |
276 |     ffi::duk_get_global_string(ctx, UDATA.as_ptr());
    |                                     ^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/util.rs:283:37
    |
283 |     ffi::duk_get_global_string(ctx, ANYMAP.as_ptr());
    |                                     ^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/ducc.rs:367:50
    |
367 |             ffi::duk_get_global_string(self.ctx, STASH_KEY.as_ptr());
    |                                                  ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/ducc.rs:390:50
    |
390 |             ffi::duk_put_global_string(self.ctx, STASH_KEY.as_ptr());
    |                                                  ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/function.rs:130:52
    |
130 |             ffi::duk_put_prop_string(ducc.ctx, -2, FUNC.as_ptr());
    |                                                    ^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
  --> /build/texlab-1.7.0-vendor/ducc/src/function.rs:86:47
   |
86 |             ffi::duk_get_prop_string(ctx, -1, FUNC.as_ptr());
   |                                               ^^^^^^^^^^^^^ expected u8, found i8
   |
   = note: expected type `*const u8`
              found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/function.rs:117:42
    |
117 |         ffi::duk_get_prop_string(ctx, 0, FUNC.as_ptr());
    |                                          ^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error[E0308]: mismatched types
   --> /build/texlab-1.7.0-vendor/ducc/src/function.rs:121:42
    |
121 |         ffi::duk_put_prop_string(ctx, 0, FUNC.as_ptr());
    |                                          ^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`
error: aborting due to 15 previous errors
For more information about this error, 
```
</details>

full log: https://logs.nix.ci/?key=nixos/nixpkgs.73839&attempt_id=c7ed5679-f27e-4199-adbd-cb2100515246

--- a/ducc/src/ducc.rs
+++ b/ducc/src/ducc.rs
@@ -364,7 +364,7 @@ impl Ducc {
             ffi::duk_push_heap_stash(self.ctx);
 
             // Generate a unique stash key.
-            ffi::duk_get_global_string(self.ctx, STASH_KEY.as_ptr());
+            ffi::duk_get_global_string(self.ctx, STASH_KEY.as_ptr() as *const _);
             let mut stash_key: ffi::duk_uarridx_t = ffi::duk_to_number(self.ctx, -1) as u32;
             ffi::duk_pop(self.ctx);
             let mut has_wrapped = false;
@@ -387,7 +387,7 @@ impl Ducc {
                 }
             }
             ffi::duk_push_uint(self.ctx, stash_key);
-            ffi::duk_put_global_string(self.ctx, STASH_KEY.as_ptr());
+            ffi::duk_put_global_string(self.ctx, STASH_KEY.as_ptr() as *const _);
 
             ffi::duk_dup(self.ctx, -2);
             ffi::duk_put_prop_index(self.ctx, -2, stash_key);

--- a/ducc/src/function.rs
+++ b/ducc/src/function.rs
@@ -83,7 +83,7 @@ pub(crate) fn create_callback<'ducc, 'callback>(
             }
 
             ffi::duk_push_current_function(ctx);
-            ffi::duk_get_prop_string(ctx, -1, FUNC.as_ptr());
+            ffi::duk_get_prop_string(ctx, -1, FUNC.as_ptr() as *const _);
             let func_ptr = ffi::duk_get_pointer(ctx, -1) as *mut Callback;
             ffi::duk_pop_n(ctx, 2);
 
@@ -114,11 +114,11 @@ pub(crate) fn create_callback<'ducc, 'callback>(
 
     unsafe extern "C" fn finalizer(ctx: *mut ffi::duk_context) -> ffi::duk_ret_t {
         ffi::duk_require_stack(ctx, 1);
-        ffi::duk_get_prop_string(ctx, 0, FUNC.as_ptr());
+        ffi::duk_get_prop_string(ctx, 0, FUNC.as_ptr() as *const _);
         Box::from_raw(ffi::duk_get_pointer(ctx, -1) as *mut Callback);
         ffi::duk_pop(ctx);
         ffi::duk_push_undefined(ctx);
-        ffi::duk_put_prop_string(ctx, 0, FUNC.as_ptr());
+        ffi::duk_put_prop_string(ctx, 0, FUNC.as_ptr() as *const _);
         0
     }
 
@@ -127,7 +127,7 @@ pub(crate) fn create_callback<'ducc, 'callback>(
             ffi::duk_require_stack(ducc.ctx, 2);
             ffi::ducc_push_c_function_nothrow(ducc.ctx, Some(wrapper), ffi::DUK_VARARGS);
             ffi::duk_push_pointer(ducc.ctx, Box::into_raw(Box::new(func)) as *mut _);
-            ffi::duk_put_prop_string(ducc.ctx, -2, FUNC.as_ptr());
+            ffi::duk_put_prop_string(ducc.ctx, -2, FUNC.as_ptr() as *const _);
             ffi::duk_push_c_function(ducc.ctx, Some(finalizer), 1);
             ffi::duk_set_finalizer(ducc.ctx, -2);
             Function(ducc.pop_ref())

--- a/ducc/src/util.rs
+++ b/ducc/src/util.rs
@@ -114,11 +114,11 @@ const ERROR_KEY: [i8; 7] = hidden_i8str!('e', 'r', 'r', 'o', 'r');
 
 unsafe extern "C" fn error_finalizer(ctx: *mut ffi::duk_context) -> ffi::duk_ret_t {
     ffi::duk_require_stack(ctx, 1);
-    ffi::duk_get_prop_string(ctx, 0, ERROR_KEY.as_ptr());
+    ffi::duk_get_prop_string(ctx, 0, ERROR_KEY.as_ptr() as *const _);
     Box::from_raw(ffi::duk_get_pointer(ctx, -1) as *mut Error);
     ffi::duk_pop(ctx);
     ffi::duk_push_undefined(ctx);
-    ffi::duk_put_prop_string(ctx, 0, ERROR_KEY.as_ptr());
+    ffi::duk_put_prop_string(ctx, 0, ERROR_KEY.as_ptr() as *const _);
     0
 }
 
@@ -154,7 +154,7 @@ pub(crate) unsafe fn push_error(ctx: *mut ffi::duk_context, error: Error) {
             ffi::duk_put_prop_string(ctx, -2, cstr!("message"));
         }
         ffi::duk_push_pointer(ctx, Box::into_raw(desc.cause) as *mut _);
-        ffi::duk_put_prop_string(ctx, -2, ERROR_KEY.as_ptr());
+        ffi::duk_put_prop_string(ctx, -2, ERROR_KEY.as_ptr() as *const _);
         ffi::duk_push_c_function(ctx, Some(error_finalizer), 1);
         ffi::duk_set_finalizer(ctx, -2);
     })
@@ -164,11 +164,11 @@ pub(crate) unsafe fn pop_error(ctx: *mut ffi::duk_context) -> Error {
     assert_stack!(ctx, -1, {
         ffi::duk_require_stack(ctx, 1);
 
-        ffi::duk_get_prop_string(ctx, -1, ERROR_KEY.as_ptr());
+        ffi::duk_get_prop_string(ctx, -1, ERROR_KEY.as_ptr() as *const _);
         let error_ptr = ffi::duk_get_pointer(ctx, -1) as *mut Error;
         ffi::duk_pop(ctx);
         ffi::duk_push_undefined(ctx);
-        ffi::duk_put_prop_string(ctx, -2, ERROR_KEY.as_ptr());
+        ffi::duk_put_prop_string(ctx, -2, ERROR_KEY.as_ptr() as *const _);
         ffi::duk_push_undefined(ctx);
         ffi::duk_set_finalizer(ctx, -2);
         if !error_ptr.is_null() {
@@ -257,11 +257,11 @@ pub(crate) unsafe fn create_heap() -> *mut ffi::duk_context {
     ffi::duk_require_stack(ctx, 1);
 
     ffi::duk_push_pointer(ctx, udata as *mut _);
-    ffi::duk_put_global_string(ctx, UDATA.as_ptr());
+    ffi::duk_put_global_string(ctx, UDATA.as_ptr() as *const _);
 
     let any_map = Box::into_raw(Box::new(AnyMap::new()));
     ffi::duk_push_pointer(ctx, any_map as *mut _);
-    ffi::duk_put_global_string(ctx, ANYMAP.as_ptr());
+    ffi::duk_put_global_string(ctx, ANYMAP.as_ptr() as *const _);
 
     ffi::duk_push_global_object(ctx);
     ffi::duk_del_prop_string(ctx, -1, cstr!("Duktape"));
@@ -273,14 +273,14 @@ pub(crate) unsafe fn create_heap() -> *mut ffi::duk_context {
 pub(crate) unsafe fn get_udata(ctx: *mut ffi::duk_context) -> *mut Udata {
     let _sg = StackGuard::new(ctx);
     ffi::duk_require_stack(ctx, 1);
-    ffi::duk_get_global_string(ctx, UDATA.as_ptr());
+    ffi::duk_get_global_string(ctx, UDATA.as_ptr() as *const _);
     ffi::duk_get_pointer(ctx, -1) as *mut Udata
 }
 
 pub(crate) unsafe fn get_any_map(ctx: *mut ffi::duk_context) -> *mut AnyMap {
     let _sg = StackGuard::new(ctx);
     ffi::duk_require_stack(ctx, 1);
-    ffi::duk_get_global_string(ctx, ANYMAP.as_ptr());
+    ffi::duk_get_global_string(ctx, ANYMAP.as_ptr() as *const _);
     ffi::duk_get_pointer(ctx, -1) as *mut AnyMap
 }
 

